docs/HIDDriverGuide.md

```diff
@@ -0,0 +1,321 @@
+4. the interface descriptor doesn't declare any IN endpoints on the device
+5. input endpoint was not open due to the limits
+of the native USB [API](https://developer.electricimp.com/api/hardware/usb).
+
+If endpoint was not open due to the limit of open Interrupt Endpoints,
+the developer may use the synchronous [HIDReport.request()](#request) call,
+which works through the endpoint 0 and isn't affected by the limitation.
+
+###### Callback Function
+
+The must accept the following parameters.
+
+| Parameter | Type | Description |
+| --------- | ---- | ----------- |
+| *error*   | String  | Non null parameter indicates an error |
+| *report*  | [HIDReport](#hidreport-class) | Read report |
+
+#### HIDReport Class
+
+This class represents HID Reports - a data packet
+that can be transferred to or from the device.
+
+##### request()
+
+Obtains HID state from the device through the Endpoint 0.
+The method doesn't return anything but may throw an exception
+if an error occurs during the transfer or if the
+control endpoint is closed.
+
+##### send()
+
+Synchronously sends the output items. The items value need to be updated
+prior to the call. Throws an exception if the endpoint is closed or an error
+occurs during the native USB API call.
+
+##### setIdleTimeMs(millis)
+
+Issues the `"Set Idle"` command through the associated endpoint. Returns nothing
+but may throw an exception, if the endpoint 0 is closed or an error
+occurs during the native USB API call.
+
+The function takes the following arguments:
+
+| Parameters name | Type | Description |
+| --------------- | ---- | ----------- |
+| *millis* | Integer | milliseconds, the idle time for the report, between 4 - 1020 ms |
+
+Refer to the section __7.2.4__ of
+[HID specification](http://www.usb.org/developers/hidpage/HID1_11.pdf) for more details.
+
+##### getInputItems()
+
+Returns an array of input items or null if no items were found in the report descriptor.
+
+##### getOutputItems()
+
+Returns an array of output items or null if no items were found in the report descriptor.
+
+##### getFeatureItems()
+
+Returns an array of feature items or null if no items were found in the report descriptor.
+
+
+#### HIDReport.Item class
+
+The class represents a single report item in an HID Report.
+It has a number of attributes:
+
+| Class Field | Type | Description |
+| ----------- | ---- | ----------- |
+| *attributes* | [HIDReport.Item.Attributes](#hidreportitemattributes-class) | HID report item tags |
+| *itemFlags*  | Integer | Defines HID Report Item value attributes. [HID Constants](#hid-constants). Should be used by application for processing the Item's data |
+| *collectionPath* | [HIDReport.Collection](#hidreportcollection) | Identifies a collection, the item belongs to |
+
+##### print(stream)
+
+Debug function, prints the report data to the specified function `stream`, for example, `server.log`.
+
+#### get()
+
+Returns the present HID report item value.
+
+#### set(value)
```

Comment:
this header is bigger than other methods at this level

Reply:
fixed